### PR TITLE
[SPARK-31484][Core][Flollowup] Use taskAttemptId in checkpoint filename

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

As suggested by https://github.com/apache/spark/pull/28255#discussion_r412619438, this patch proposes to use taskAttemptId in checkpoint filename, instead of stageAttemptNumber + attemptNumber.

### Why are the changes needed?

To simplify checkpoint simplified and unique.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by viirya** [comment]
cc @cloud-fan @xuanyuanking @dongjoon-hyun

**Review by xuanyuanking — APPROVED** [review]
LGTM

**Review by Ngone51 — APPROVED** [review]

**Comment by viirya** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala [hunk 1]
```diff
@@ -199,8 +199,7 @@ private[spark] object ReliableCheckpointRDD extends Logging {
 
     val finalOutputName = ReliableCheckpointRDD.checkpointFileName(ctx.partitionId())
     val finalOutputPath = new Path(outputDir, finalOutputName)
-    val tempOutputPath = new Path(outputDir,
-      s".$finalOutputName-attempt-${ctx.stageAttemptNumber()}-${ctx.attemptNumber()}")
+    val tempOutputPath = new Path(outputDir, s".$finalOutputName-attempt-${ctx.taskAttemptId()}")
```
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala:202]: Ya. It looks simpler.

**Review by dongjoon-hyun — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala:202]: BTW, do we need `-attempt-`?

**Review by HyukjinKwon — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala (hunk 1 above)
Comment (cloud-fan) [reply core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala:202]:
```
private[spark] class TaskDescription(
    val taskId: Long,
    val attemptNumber: Int,
...
```
I believe `taskAttempId` is some kind of historic name in `TaskContext`.

**Review by cloud-fan — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by viirya** [comment]
Thanks!

**Pull request closed without merging**